src/test/java/integration/IntegrationTest.java

```diff
@@ -3,563 +3,1428 @@
+        testProjectGetFiles();
+        testFileMove(ws2, file1, "newPathWS2", new WSManager[]{ws1});
+        testProjectGetFiles();
+        testFileMove(ws2, file2, "newPathWS2", new WSManager[]{});
+        testProjectGetFiles();
+        testFileMoveInvalid();
+
+        testFileChange(ws1, file1, new WSManager[]{ws2}); // Test User1 changing File1
+        testFilePull(); // Should have 3 changes on File1, 1 change on File2
+        testFileChange(ws2, file2, new WSManager[]{}); // Test User2 changing File2
+        testFilePull(); // Should have 3 changes on File1, 2 change on File2
+
+        testFileRename(ws1, file1, "_newNameWS1", new WSManager[]{ws2});
+        testProjectGetFiles();
+        testFileRename(ws2, file1, "_newNameWS2", new WSManager[]{ws1});
+        testProjectGetFiles();
+        testFileRename(ws2, file2, "_newNameWS2", new WSManager[]{});
+        testProjectGetFiles();
+
+        testFileChange(ws1, file1, new WSManager[]{ws2}); // Test User1 changing File1
+        testFilePull(); // Should have 4 changes on File1, 2 change on File2
+        testFileChange(ws2, file2, new WSManager[]{}); // Test User2 changing File2
+        testFilePull(); // Should have 4 changes on File1, 3 change on File2
+
+        testProjectGrantPermission(ws1, proj1, user2ID, "admin", new WSManager[]{ws2});
+        testProjectGrantPermission(ws2, proj1, user3ID, "read", new WSManager[]{ws1});
+
+        testProjectDeleteInvalid(); // Test this here to make sure even admins don't have deletion rights.
+
+        testProjectRevokePermissionInvalid();
+        testProjectRevokePermission(ws2, proj1, user3ID, new WSManager[]{ws1});
+        testProjectRevokePermission(ws2, proj1, user2ID, new WSManager[]{ws1});
+        testFileChange(ws1, file1, new WSManager[]{}); // Test User1 changing File1; Should have 5 changes on File1, 3 change on File2
+        testProjectGrantPermission(ws1, proj1, user2ID, "write", new WSManager[]{ws2});
+        testProjectSubscribe(ws2, proj1);
+
+        testProjectUnsubscribe(ws2, proj1);
+        testFileChange(ws1, file1, new WSManager[]{}); // Test User1 changing File1; Should have 6 changes on File1, 3 change on File2
+        testProjectSubscribe(ws2, proj1);
+
+        testFileDeleteInvalid();
+        testFileDelete(ws2, file1, new WSManager[]{ws1});
+        testFileDelete(ws2, file2, new WSManager[]{});
+        testProjectGetFiles(); // should have 0 files each
+
+        testProjectLookup(ws1, user1ID, proj1);
+        testProjectLookup(ws2, user2ID, proj2);
+
+        testProjectDelete(ws1, proj1, new WSManager[]{ws2});
+        testProjectDelete(ws2, proj2, new WSManager[]{});
+
+        testUserProjects();
+
+        testUserDelete(ws1, user1ID, user1Pass);
+        testUserDelete(ws2, user2ID, user2Pass);
+
+        testUserLogin(ws3, user3ID, user3Pass);
+        testUserDelete(ws3, user3ID, user3Pass);
+
+        Thread.sleep(1000);
```

Comment:
That's actually from the old integration tests; It's to make sure that we have all the requests done, and the handlers deleted before moving on. There were some issues with taking it out in the old version of the integration tests, and thus, it's still there.